src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,410 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import * as d3 from 'd3';
+
+import { getCssToString, zoomFunction, distinctColors, transformDataIntoMapArray, randomId, getMinMaxValue } from './utils';
+import injectData from '../injectData';
+import exportableToPng from '../exportableToPng';
+
+const styles = {
+    divContainer: {
+        overflow: 'hidden',
+        position: 'relative',
+    },
+    tooltip: {
+        marginTop: 900,
+    },
+    vertical: {
+        marginTop: 60,
+        height: 190,
+        pointerEvents: 'none',
+    },
+    legend: {
+        position: 'relative',
+    },
+    legendItem: {
+        marginTop: 2,
+    },
+    legentItemText: {
+        marginLeft: 5,
+    },
+    legendButton: {
+        height: 15,
+        width: 15,
+    },
+};
+
+class Streamgraph extends PureComponent {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            width: 800,
+            height: 300
+        }
+
+        this.uniqueId = randomId(10);
+        this.divContainer = "divContainer" + this.uniqueId;
+        this.svgContainer = "svgContainer" + this.uniqueId;
+        this.anchor = "anchor" + this.uniqueId;
+
+        this.xAxisScale;
+        this.yAxisScale;
+        this.xAxis;
+        this.yAxisL;
+        this.yAxisR;
+        this.gx;
+        this.gyr;
+        this.gyl;
+        this.streams;
+
+        //this.draft = draft.bind(this);
+        this.zoomFunction = zoomFunction.bind(this);
+    }
+
+    setGraph() {
+        let [valuesObjectsArray, valuesArray, minDate, maxDate, nameList] = transformDataIntoMapArray(this.props.formatData);
+        const divContainerWidth = document.getElementById(this.divContainer).clientWidth;
```

Comment:
This don't use the `document` API directly from a React component.
To manipulate DOM elements, you can use the [React refs API](https://reactjs.org/docs/refs-and-the-dom.html).